doit.go

```diff
@@ -40,26 +44,115 @@ func (c *LiveConfig) GetGodoClient() *godo.Client {
 	return godo.NewClient(oauthClient)
 }
 
+func sshConnect(user string, host string, method ssh.AuthMethod) (err error) {
+	sshc := &ssh.ClientConfig{
+		User: user,
+		Auth: []ssh.AuthMethod{method},
+	}
+	conn, err := ssh.Dial("tcp", host, sshc)
+	if err != nil {
+		return err
+	}
+
+	session, err := conn.NewSession()
+	if err != nil {
+		return err
+	}
+
+	session.Stdout = os.Stdout
+	session.Stderr = os.Stderr
+	session.Stdin = os.Stdin
+
+	defer session.Close()
+
+	modes := ssh.TerminalModes{
+		ssh.ECHO: 1,
+	}
+
+	var (
+		termWidth, termHeight int
+	)
+	fd := os.Stdin.Fd()
+	if term.IsTerminal(fd) {
+		oldState, err := term.MakeRaw(fd)
+		if err != nil {
+			return err
+		}
+
+		defer term.RestoreTerminal(fd, oldState)
+
+		winsize, err := term.GetWinsize(fd)
+		if err != nil {
+			termWidth = 80
+			termHeight = 24
+		} else {
+			termWidth = int(winsize.Width)
+			termHeight = int(winsize.Height)
+		}
+	}
+
+	if err := session.RequestPty("xterm", termWidth, termHeight, modes); err != nil {
+		session.Close()
+		return err
+	}
+	if err == nil {
+		err = session.Shell()
+	}
+	if err != nil {
+		return err
+	}
+
+	err = session.Wait()
+	if err != nil && err != io.EOF {
+		// Ignore the error if it's an ExitError with an empty message,
+		// this occurs when you do CTRL+c and then run exit cmd which isn't an
+		// actual error.
+		waitMsg, ok := err.(*ssh.ExitError)
+		if ok && waitMsg.Msg() == "" {
+			return nil
+		}
+
+		return err
+	}
+	return err
+}
+
 // SSH creates a ssh connection to a host.
-func (c *LiveConfig) SSH(user, host string, options []string) Runner {
+func (c *LiveConfig) SSH(user, host, keyPath string, port int) (err error) {
 	logrus.WithFields(logrus.Fields{
 		"user": user,
 		"host": host,
 	}).Info("ssh")
 
-	sshHost := fmt.Sprintf("%s@%s", user, host)
+	sshHost := fmt.Sprintf("%s:%d", host, port)
 
-	args := []string{sshHost}
-	for _, o := range options {
-		args = append(args, "-o", o)
+	// Key Auth
+	key, err := ioutil.ReadFile(keyPath)
+	if err != nil {
+		return err
 	}
-
-	cmd := exec.Command("ssh", args...)
-	cmd.Stdin = os.Stdin
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
-
-	return cmd
+	privateKey, err := ssh.ParsePrivateKey(key)
+	if err != nil {
+		return err
+	}
+	if err := sshConnect(user, sshHost, ssh.PublicKeys(privateKey)); err != nil {
+		// Password Auth if Key Auth Fails
```

Comment:
idt its a valid assumption that `err != nil` means you couldn't auth due to password needs. Instead of needing a password, use the `ssh-agent` provider in https://godoc.org/golang.org/x/crypto/ssh/agent. likely the `keyPath` thingy won't be needed as well